### PR TITLE
Ember 2.0 support

### PR DESCRIPTION
**Pull request by GavinJoyce**
closes https://github.com/eviltrout/ember-performance/issues/51

This updates the tests to run on Ember 1.11 -> Ember 2.0. I've implemented a new [RenderTemplateTestClient](https://github.com/GavinJoyce/ember-performance/blob/gj/ember-2.0/test-client/test-client.js#L356-L392) which allows for [simple tests](https://github.com/GavinJoyce/ember-performance/blob/gj/ember-2.0/tests/render-list/index.js) such as:

```javascript
(function() {
  var template = "hi {{name}}";

  RenderTemplateTestClient.run({
    name: 'Render simple attribute',

    setup: function() {
      this.setupTemplateTest(template, { name: 'Ben' });
    },

    reset: function() {
      this.hideComponent();
    },

    test: function() {
      this.showComponent();
    }
  });
})();
```

I've dropped versions earlier than Ember 1.11 as they would require separate templates.

### CODE REVIEW

File: tests/render-complex-list/index.js [hunk 1]
```diff
@@ -1,55 +1,36 @@
...
+(function() {
+  var template =
     "<ul>" +
-      "{{#each view.listItems}}" +
-        "<li>{{buffer-render data=a}} {{buffer-render data=b}} {{buffer-render data=c}} {{buffer-render data=d}}</li>" +
-        "<li>{{buffer-render data=a}} {{buffer-render data=b}} {{buffer-render data=c}} {{buffer-render data=d}}</li>" +
-        "<li>{{component-render a=a b=b c=c d=d}}</li>" +
-        "<li>{{component-render a=a b=b c=c d=d}}</li>" +
+      "{{#each data.items as |item|}}" +
+        "<li>{{buffer-render data=a}} {{buffer-render data=item.b}} {{buffer-render data=item.c}} {{buffer-render data=item.d}}</li>" +
```
Comment (GavinJoyce) [inline tests/render-complex-list/index.js]: this should be `data=item.a`

**Comment by GavinJoyce** [comment]
I've gathered results from before and after this PR to ensure that we're still capturing the slowdown and subsequent speed improvements in initial render performance:

https://docs.google.com/spreadsheets/d/15MPfzCncD_PvM-nmfdrA-_UsUzNpAVV9pFkt3wU5rBo/edit?usp=sharing

The new tests seem to be capturing this correctly (master on the left, https://github.com/eviltrout/ember-performance/pull/52 on the right):

image

**Comment by eviltrout** [comment]
This looks excellent, I will merge now :)

**Pull request merged**